perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -1269,7 +1271,18 @@ public String validatePreferredEmailChange(PerunSession sess, User user, String
 	}
 
 	@Override
-	public void changeNonAuthzPassword(PerunSession sess, String i, String m, String password, String lang) throws UserNotExistsException, LoginNotExistsException, PasswordChangeFailedException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException, PasswordStrengthException {
+	public void checkPasswordResetRequestIsValid(PerunSession sess, String i, String m) throws UserNotExistsException, PasswordResetLinkExpiredException, PasswordResetLinkNotValidException {
```

Comment:
I think we could add authorization to this method, allowing only SELF and PERUNADMIN to perform the operation. However, I am not sure if it is needed. What do you think @zlamalp ?

Reply:
This will be called by non-authorized user, so we can't check the caller against the user inside the request. For now it has to be without authorization.

Reply:
ok